services/app-web/src/gqlHelpers/apolloQueryWrapper.ts

```diff
@@ -67,12 +67,18 @@ function wrapApolloResult<ResultType extends WrappableApolloResultsType>(
             },
         }
     }
+    class SkipError extends Error {
+        constructor(message: string) {
+          super(message);
+          this.name = "SKIPPED";
+        }
+      }
 
     return {
         ...queryResult,
         result: {
             status: 'ERROR',
-            error: new Error('UNEXPECTED APOLLO BEHAVIOR, NO DATA'),
+            error: new SkipError('Skipped query'),    //this happens when  uery is skipped, not unexpected behavior
```

Comment:
when are queries skipped?

Reply:
I skip the query for fetching a rate when submission type loads on the `/submissions/new` here https://github.com/Enterprise-CMCS/managed-care-review/pull/2276/files#diff-fa4eb2164c7d663a948b698c67738aee06da93091bf18d2f99854a4112cae011R56.  This relies on this wrapperd https://github.com/Enterprise-CMCS/managed-care-review/pull/2276/files#diff-6d057babf0fd32dc9820312d200c0c9ff11e28b40abb3a99e707d3c55f3e1e92R121-R129 . 

I wanted to keep HPP code together in the hook so its much easier to remove and feature flag. But, because react hooks must use the exact same order in every component render and `if` statements cannot be used, I  also needed some other way to still conditionally call the `fetch` request that is meant to happen on load in all cases but the case when you start a new submission. I could have let the API hit the api and just return 404 but we had other UI handling latching in the 404 to display not found so this seemed the most direct way to avoid the request instead of adding more UI logic where we already have a lot going on.